### DISCUSSION TITLE
Bugs when converting set-outputs steps

### DISCUSSION THREAD

**Discussion by k-doering-NOAA** [community/community#42790] [Actions] [answered]
### Select Topic Area

Bug

### Body

@e-gugliotti-NOAA and I are encountering some potential bugs when trying to [patch our github actions currently using set-output](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/#patching-your-actions-and-workflows):
1. After making the change from set-output to environmental files, the output from 1 job is not properly being passed to a second job in the workflow. The job fails on [runs-on: of the second job](https://github.com/nmfs-fish-tools/ghactions4r/blob/e7ed07d2ce071c3070d69d56df984cadba1e2d50/.github/workflows/r-cmd-check.yml#L51)
2. After making the change from set-output to environmental files, the output from 1 step fails to be passed for use in a windows powershell command and [the run fails at the powershell command](https://github.com/nmfs-stock-synthesis/workflows/blob/85f79236758b28469b756acd5d3c92e9615b5c25/.github/workflows/build-ss3.yml#L104); however, similar steps work fine on ubuntu and mac.

Reading through the documentation, it seems like these scenarios should work, and that we are specifying it correctly, but we could be wrong. Any help solving these mysteries would be appreciated!

**Comment by airtower-luna** [comment, marked as answer]
> After making the change from set-output to environmental files, the output from 1 job is not properly being passed to a second job in the workflow.

The part where you create the matrix [escapes the data](https://github.com/nmfs-fish-tools/ghactions4r/blob/e7ed07d2ce071c3070d69d56df984cadba1e2d50/.github/workflows/r-cmd-check.yml#L44-L46) as you needed to do with the `set-output` command. This isn't necessary any more and will break your output string, because the escapes will be part of the value when you use it). Instead, you need to create a multi-line output using the [same format as for a multi-line environment variable](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings). I have an example [here](https://github.com/airtower-luna/hello-github-actions/blob/e3d5aa3790dc41103969c7cc77e18c60abf8b32f/.github/workflows/multiline-output.yaml) (using Python to generate a random delimiter).

> After making the change from set-output to environmental files, the output from 1 step fails to be passed for use in a windows powershell command and [the run fails at the powershell command](https://github.com/nmfs-stock-synthesis/workflows/blob/85f79236758b28469b756acd5d3c92e9615b5c25/.github/workflows/build-ss3.yml#L104);

Most likely the output isn't set correctly. The examples for how to set outputs use Unix shell syntax, which doesn't work in Powershell (which is the default shell for `run` steps on Windows). The simplest solution is to set `shell: bash` for the step where you create the outputs, as long as you don't actually need Powershell there. The alternative is to convert it to the equivalent Powershell syntax (which I can't tell you how to do).

**Reply by k-doering-NOAA** [reply]
@airtower-luna , thank you for the helpful response!
For 1, you were correct that escaping the data was not necessary, so deleting those 3 lines and using the formatting for multiline outputs was what was needed to get the workflow running. 
For 2, we discovered that `echo "whatever" >> $env: GITHUB_OUTPUT` needed to be used with powershell rather than `echo "whatever" >> $GITHUB_OUTPUT` within the workflow.

**Reply by airtower-luna** [reply]
Glad I could help, and thanks for sharing your solution for Powershell!
